crates/core/src/bc_protocol/ping.rs

```diff
@@ -28,4 +28,40 @@ impl BcCamera {
 
         Ok(())
     }
+
+    /// Send a ping repeatedly
+    pub fn ping_every(&self, milis: u64) -> Result<()> {
+        let wait = std::time::Duration::from_millis(milis);
+
+        let connection = self.connection.as_ref().expect("Must be connected to ping");
+        let sub_ping = connection.subscribe(MSG_ID_PING)?;
+
+        loop {
+            let ping = Bc {
+                meta: BcMeta {
+                    msg_id: MSG_ID_PING,
+                    channel_id: self.channel_id,
+                    msg_num: self.new_message_num(),
+                    stream_type: 0,
+                    response_code: 0,
+                    class: 0x6414,
+                },
+                body: BcBody::ModernMsg(ModernMsg {
+                    ..Default::default()
+                }),
+            };
+
+            sub_ping.send(ping)?;
+
+            let reply = sub_ping.rx.recv_timeout(RX_TIMEOUT)?;
+
+            if reply.meta.response_code != 200 {
+                return Err(Error::UnintelligibleReply {
+                    reply,
+                    why: "Camera responded with an error status code to the ping",
+                });
+            }
+            std::thread::sleep(wait);
```

Comment:
Why should `ping_every` form part of the public API for the core crate? Seems like a loop the application should do.

If this needs to be part of the public API, call ping from this method, don't copy paste.

Reply:
I think perhaps remove this, it was part of my testing because the UDP camera needs a new message from the client regularly or it will disconnect. This is now done through the keep alive message in `BcConnection`

Reply:
Removed